common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -105,6 +105,8 @@ def target_branch
       sig { returns(T::Array[DependencyFile]) }
       def files
         @files ||= T.let(fetch_files, T.nilable(T::Array[DependencyFile]))
+        @files.each { |f| f.job_directory = directory }
```

Comment:
They will be set each time `FileFetcher#files` is used, should it be memoized instead?